test/integration/main_test.go

```diff
@@ -0,0 +1,412 @@
+[
+  "https://cloudresourcemanager.googleapis.com/v1/projects/%s"
+]
+
+gcloud projects list --uri --sort-by=projectId --limit=1
+https://cloudresourcemanager.googleapis.com/v1/projects/%s
+------Successfully Completed Commands------
+- gcloud projects list --format json --uri --sort-by=projectId --limit=1
+- gcloud projects list --uri --sort-by=projectId --limit=1
+`, cfg.ProjectID, cfg.ProjectID),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"tool", "cleanup",
+				"-path", filePath,
+			}
+			if tc.verbose {
+				args = append(args, "-verbose")
+			}
+
+			_, stdout, stderr := testPipeAndRun(ctx, t, args)
+
+			if strings.TrimSpace(tc.wantOutput) != strings.TrimSpace(stdout.String()) {
+				t.Errorf("Process(%+v) output response got %q, want %q)", tc.name, stdout.String(), tc.wantOutput)
+			}
+			if stderr.String() != "" {
+				t.Errorf("Process(%+v) got unexpected error: %q)", tc.name, stderr.String())
+			}
+		})
+	}
+}
+
+func TestToolValidate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	filePath := testWriteDataToFile(t, toolReqData, "tool.yaml")
+	wantOutput := "Successfully validated tool request"
+
+	args := []string{
+		"tool", "validate",
+		"-path", filePath,
+	}
+
+	_, stdout, stderr := testPipeAndRun(ctx, t, args)
+
+	if strings.TrimSpace(wantOutput) != strings.TrimSpace(stdout.String()) {
+		t.Errorf("Output response got %q, want %q)", stdout.String(), wantOutput)
+	}
+	if stderr.String() != "" {
+		t.Errorf("Got unexpected error: %q)", stderr.String())
+	}
+}
+
+// testAddedPolicyBindings is a helper function that returns the IAM bindings
+// of matched condition title in the cfg. It also removes them from the project
+// IAM policy as cleanup.
+func testAddedPolicyBindings(ctx context.Context, tb testing.TB, cfg *config) (result []*iampb.Binding) {
+	tb.Helper()
+
+	getIAMReq := &iampb.GetIamPolicyRequest{
+		Resource: fmt.Sprintf("projects/%s", cfg.ProjectID),
+		Options: &iampb.GetPolicyOptions{
+			RequestedPolicyVersion: 3,
+		},
+	}
+	backoff := retry.WithMaxRetries(cfg.QueryRetryLimit, retry.NewConstant(cfg.QueryRetryWaitDuration))
+
+	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+		p, err := projectClient.GetIamPolicy(ctx, getIAMReq)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to get IAM policy: %w", err))
+		}
+		var bs []*iampb.Binding
+		for _, b := range p.Bindings {
+			if b.Condition != nil && b.Condition.Title == cfg.ConditionTitle {
+				result = append(result, b)
+				continue
+			}
+			bs = append(bs, b)
+		}
+		p.Bindings = bs
+		setIAMReq := &iampb.SetIamPolicyRequest{
+			Resource: fmt.Sprintf("projects/%s", cfg.ProjectID),
+			Policy:   p,
+		}
+		if _, err := projectClient.SetIamPolicy(ctx, setIAMReq); err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to set IAM policy: %w", err))
+		}
```

Comment:
Smart - you're piggybacking the cleanup here. But could you rename the func to be `testGetAndResetBindings`